tower-http/src/validate_request.rs

```diff
@@ -379,25 +379,24 @@ where
             .headers()
             .get_all(header::ACCEPT)
             .into_iter()
-            .flat_map(|header| {
-                header
-                    .to_str()
-                    .ok()
-                    .into_iter()
-                    .flat_map(|s| s.split(",").map(|typ| typ.trim()))
-            })
+            .filter_map(|header| header.to_str().ok())
             .any(|h| {
-                h.parse::<Mime>()
+                MimeIter::new(&h)
                     .map(|mim| {
-                        let typ = self.header_value.type_();
-                        let subtype = self.header_value.subtype();
-                        match (mim.type_(), mim.subtype()) {
-                            (t, s) if t == typ && s == subtype => true,
-                            (t, mime::STAR) if t == typ => true,
-                            (mime::STAR, mime::STAR) => true,
-                            _ => false,
+                        if let Ok(mim) = mim {
+                            let typ = self.header_value.type_();
+                            let subtype = self.header_value.subtype();
+                            match (mim.type_(), mim.subtype()) {
+                                (t, s) if t == typ && s == subtype => true,
+                                (t, mime::STAR) if t == typ => true,
+                                (mime::STAR, mime::STAR) => true,
+                                _ => false,
+                            }
+                        } else {
+                            false
                         }
                     })
+                    .reduce(|acc, mim| acc || mim)
```

Comment:
Do we not want all to be true here?

Reply:
We want to find one matching accept, right?